### PR TITLE
Feature/unr 4576 fix the tests

### PR DESCRIPTION
**Pull request by samiwh**
Read in commit order for an easier time.

The edit count is so high because it uncomments the old tests. Skip the commit uncommenting the tests to avoid seeing said edits. Can turn this into two dependent PRs if this makes it much easier.

##Description
Fix the tests.
Component sets need to be added to anything working with op lists.
This includes all tests that use op lists to create view deltas.
Minor drive-by changes on tests that are being touched for other reasons.

Removed a couple of tests that don't really make sense.

Async asset loading currently works by coincidence. We create op lists representing an owned version of the changes that have occurred while waiting for the asset to load but we don't maintain a view. However the oplists are no longer possible to create correctly without a view and component set data. This currently works because we don't completely correctly read the op list either and actually read it as if it was the old op lists or the view delta intermediate format. The same goes for view delta -> oplist conversations consumed by spatial receiver.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 1]
```diff
@@ -2739,7 +2739,10 @@ void USpatialReceiver::QueueAuthorityOpForAsyncLoad(const Worker_ComponentSetAut
 {
 	EntityWaitingForAsyncLoad& AsyncLoadEntity = EntitiesWaitingForAsyncLoad.FindChecked(Op.entity_id);
 
-	AsyncLoadEntity.PendingOps.SetAuthority(Op.entity_id, Op.component_set_id, static_cast<Worker_Authority>(Op.authority));
+	// todo this needs to be changed as it abuses the authority ops in a way that happens to work here.
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp:2742]: I don't understand the reasoning here - but okay if it'll be resolved with the pending proper component set usage

File: SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ViewDeltaTest.cpp [hunk 2]
```diff
@@ -230,6 +230,67 @@ VIEWDELTA_TEST(GIVEN_entity_and_auth_component_in_view_WHEN_authority_lost_THEN_
...
+
+	return true;
+}
+
+VIEWDELTA_TEST(GIVEN_entity_with_no_components_WHEN_authority_gained_with_non_empty_component_data_THEN_component_added_to_view)
+{
+	ViewDelta InputDelta;
+	EntityView InputView;
+	FComponentSetData ComponentSetData = GetTestComponentSetData();
+	ComponentData TestComponentData = CreateTestComponentData(TestComponentId, TestComponentValue);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ViewDeltaTest.cpp:268]: Worth using a component set with more than a single component id so we can
1/ validate multiple components can get added
2/ there's not confusion between component set id and component id (as they're currently always the same in our tests)

File: SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ViewDeltaTest.cpp [hunk 3]
```diff
@@ -230,6 +230,67 @@ VIEWDELTA_TEST(GIVEN_entity_and_auth_component_in_view_WHEN_authority_lost_THEN_
 	return true;
 }
 
+VIEWDELTA_TEST(GIVEN_entity_and_component_in_view_WHEN_authority_gained_with_no_component_data_THEN_component_removed_from_view)
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ViewDeltaTest.cpp:233]: could be worth having the cases like 2 components to 1, 1 component removed but a new one added, etc.- it feels like there's a lot of edge cases to this logic

**Review by jacquese — APPROVED** [review]
nice

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp:2742]: It probably shouldn't be a todo. It's fixed by the actor system filtering.

Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp:2742]: It gets solved with the actor system. Currently the async asset loading tries to record all ops that get sent to it. However when we upgraded to 15 we didn't change how anything using "ops" interacted with authority delegations. This is sort of cancelled out by also not changing how we convert view deltas to op lists for legacy code. So this function call is one that would normally be cause for concern.

So the fixes are to stop using ops for things that aren't really ops, or to turn this into a completeness filter, which is what actor systems will eventually do.

File: SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/CommandRetryHandlerTest.cpp [hunk 4]
```diff
@@ -23,7 +23,15 @@ const uint32 TestNumOfEntities = 10;
 const float TimeAdvanced = 5.f;
 OpList EmptyOpList = {};
 constexpr FRetryData TWO_RETRIES = { 2, 0, 0.1f, 5.0f, 0 };
-} // namespace SpatialGDK
+
+const Worker_ComponentSetId TestComponentSetId = 3;
+FComponentSetData GetComponentSetData()
```
[the remark this answers is not in the document]
Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/CommandRetryHandlerTest.cpp:28]: I originally thought it was too many brackets but actually it's probably overall easier to read if it's a constant.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: Changed it to point to the actor system asset filtering ticket.

File: SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ViewDeltaTest.cpp (hunk 2 above)
Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ViewDeltaTest.cpp:258]: Good points. Added.
